src/pages/randomCharacter/RandomCharacter.tsx

```diff
@@ -1,14 +1,82 @@
-import React from 'react';
-import NavBar from '../../components/navbar/NavBar';
+import React, {useState, useEffect, useCallback} from 'react';
+import CharacterCard from '../../components/characterCard/CharacterCard';
+import Layout from '../../components/layout/Layout';
 import style from './RandomCharacter.module.scss'
+import { ICharacter } from '../../services/types';
+import { getRandom } from '../../services/randomiser';
+import { useSearchParams } from 'react-router-dom';
+import Modal from '../../components/Modal/Modal';
 
-const RandomCharacter = () => {
-    return (
-        <div className={style.randomCharacter_page_outer_wrapper}>
-            <NavBar/>
-            <h1>Hier should be some random character</h1>
-        </div>
+export const BASE_URL = "https://the-one-api.dev/v2/character";
+
+
+const RandomCharacter:React.FC = () => {
+
+    const [isModalVisible, setIsModalVisible] = useState(false);
+    const [character, setCharacter] = useState<ICharacter>();
+
+    const searchedCharacter = (charName:string): void => {
+        setCharacter(characters.find(({ name }) => name.toLowerCase() === charName.toLowerCase()));
+    };
+
+    const someNumber = getRandom();
+
+    const [characters, setCharachters] = useState<ICharacter[]>([]);
+    const [randomCharacter, setRandomCharacter] = useState<ICharacter>(characters[someNumber]);
+
+    const [search, setSearch] = useSearchParams();
+    const [paramsId, setParamsId] = useState(search.get("id"));
+
+    const getRandomCharacter = useCallback(() => {
+        if(characters.length) {
+            setRandomCharacter(characters[someNumber]);
+            setSearch( {id: randomCharacter._id} );
+            }
+        }
+        , [characters, randomCharacter, setSearch]
     );
+
+    useEffect(() => {
+        if(paramsId) {
+            const char = characters.find(({_id}) => _id === paramsId);
+            if(char) {
+                setRandomCharacter(char);
+            }
+        }
+    }, [characters, paramsId]);
+
+    useEffect(() => {
+        fetch(BASE_URL, {
```

Comment:
only if !id in your url. 
no need to fetch all billion chars and loop it for every click, leave it to back-end id request.